jetty-websocket/javax-websocket-common/src/main/java/org/eclipse/jetty/websocket/javax/common/JavaxWebSocketFrameHandler.java

```diff
@@ -275,14 +254,23 @@ public void onOpen(CoreSession coreSession) throws Exception
             }
             catch (Throwable cause)
             {
-                throw new WebSocketException(endpointInstance.getClass().getName() + " OPEN method error: " + cause.getMessage(), cause);
+                Exception wse = new WebSocketException(endpointInstance.getClass().getName() + " OPEN method error: " + cause.getMessage(), cause);
+
+                // TODO This feels like double handling of the exception? Review need for futureSession
+                futureSession.completeExceptionally(wse);
+                throw wse;
             }
         }
 
         container.addBean(session, true);
         futureSession.complete(session);
     }
 
+    /**
+     * @see #onFrame(Frame,Callback)
+     */
+    public final void onFrame(Frame frame) {}
```

Comment:
The thinking is that once you override the async onFrame method, then the sync onFrame will never be called.  Thus making it final makes it clear to any derived classes that they have to extend the async because they can't extend the sync.

Reply:
Why keeping the sync methods around at all?
This is new code, it's not that we need to keep sync versions of methods when we have decided to go async.
Remove them all and everything will be simpler.
If you're worried of the tests, it's a simple factoring click in IJ.

Reply:
They are convenience methods that avoid having lots of duplicate try catches throughout the code.   So I'd like to have them somewhere.....  but perhaps you're right... will ponder.

Reply:
There are only 3 non tests implementations of FrameHandler, so I think we can remove the Adaptor (Actually move it to a test class).